package/libskylark/src/libskylark.h

```diff
@@ -0,0 +1,152 @@
+  E_PUB_CONNECTION_ERROR = 11,
+  E_PUB_READ_ERROR = 12,
+  E_BAD_HTTP_HEADER = 13,
+  E_NO_ROVER_POS_FOUND = 14,
+  E_UNAUTHORIZED_CLIENT = 15,
+  E_MAX_ERROR = 16,
+} RC;
+
+const char *client_strerror(RC code);
+void log_client_error(RC code);
+
+/**
+ *  Constant definitions: libcurl configuration and Skylark status codes.
+ */
+
+/** Defined Skylark HTTP codes. */
+typedef enum {
+  STATUS_PUT_OK = 200,
+  STATUS_GET_OK = 202,
+  STATUS_NO_HEADER = 400,
+  STATUS_BAD_HEADER = 400,
+  STATUS_NO_ROVER_POSITION = 404,
+  STATUS_SOURCE_EXIT = 400,
+  STATUS_BASE_EXIT = 500,
+  STATUS_PATH_NOT_FOUND = 404,
+  STATUS_METHOD_NOT_FOUND = 400,
+} response_status_t;
+
+// Header definitions
+#define STREAM_ENCODING "Transfer-Encoding: chunked"
+#define SBP_V2_ACCEPT_TYPE "Accept: application/vnd.swiftnav.broker.v1+sbp2"
+#define SBP_V2_CONTENT_TYPE \
+  "Content-Type: application/vnd.swiftnav.broker.v1+sbp2"
+#define DEVICE_UID_HEADER_FMT "Device-Uid: %s"
+#define USER_AGENT "libskylark-agent/1.0"
+
+/**
+ *  Constant definitions: Skylark - Piksi settings groups and boilerplate.
+ */
+
+#define SETTINGS_SKYLARK_GROUP "skylark"
+#define SETTINGS_SKYLARK_ENABLE "enable"
+#define SETTINGS_SKYLARK_URL "url"
+
+#define FILE_SKYLARK_UUID "/cfg/device_uuid"
+#define UUID4_SIZE 37
+
+/**
+ *  Constant definitions: Common type definitions around configuration and
+ *  header files.
+ */
+
+/** Callback functions for reading/writing pipe file descriptors. */
+typedef size_t write_callback_fn(char *ptr, size_t size, size_t nmemb,
+                                 void *userdata);
+typedef size_t read_callback_fn(char *buffer, size_t size, size_t nitems,
+                                void *instream);
+
+/** Structure containing Skylark client configuration. */
+typedef struct {
+  char endpoint_url[BUFSIZE]; /**< Request endpoint */
+  char device_uuid[BUFSIZE]; /**< Device UUID (UUID4) */
+  char device_header[BUFSIZE];
+  int num_retries;    /**< Number of retries */
+  int retry_delay;    /**< Delay between retries (seconds) */
+  int retry_max_time; /**< Maximum time to keep retrying (seconds) */
+  int fd;                 /**< Pipe file descriptor */
+} client_config_t;
+
+void log_client_config(const client_config_t *config);
+
+int client_config_compare(const client_config_t *a, const client_config_t *b);
+
+/**
+ *  Settings and HTTP client configuration setup.
+ */
+
+RC get_device_uuid(char *uuid);
+RC get_device_header(const char *uuid, char *uuid_header);
+RC init_config(client_config_t *config);
+
+RC setup_globals(void);
+void teardown_globals(void);
+
+/**
+ *  Download (i.e., rover) processes.
+ */
+
+size_t download_callback(char *ptr, size_t size, size_t nmemb, void *userdata);
+
+RC download_process(client_config_t *config, write_callback_fn cb,
+                    bool verbose);
+
+/**
+ *  Upload processes, for base stations and reference station processing.
+ */
+
+size_t upload_callback(char *buffer, size_t size, size_t nitems,
```

Comment:
There seem to be too many prototypes in here - things like upload_callback and download_callback should be local/static to the library and not exported. Only export here what you intend others to call or use - everything else should be hidden. If you need things for testing, then create an internal header file.